File: app/src/main/java/com/softech/code/recipe/database/FavoriteDAO.java
package com.softech.code.recipe.database;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;


import com.softech.code.recipe.model.MealFavorite;

import java.util.List;

@Dao
public interface FavoriteDAO {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(MealFavorite meal);
    
    @Query("SELECT * FROM favorite")
    List<MealFavorite> select();

    @Query("DELETE FROM favorite WHERE strMeal = :name")
    void delete(String name);
    
    @Query("SELECT * FROM favorite WHERE strMeal = :name")
    boolean isFavorite(String name);
}

File: app/src/main/java/com/softech/code/recipe/database/FavoriteDAO.kt
package com.softech.code.recipe.database

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.softech.code.recipe.model.MealFavorite

@Dao
interface FavoriteDAO {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(meal: MealFavorite?): Long

    @Query("SELECT * FROM favorite")
    fun select(): List<MealFavorite?>?

    @Query("DELETE FROM favorite WHERE strMeal = :name")
    fun delete(name: String?)

    @Query("SELECT * FROM favorite WHERE strMeal = :name")
    fun isFavorite(name: String?): Boolean
}